zomes/blog/code/src/main_capability.rs

```diff
@@ -3,6 +3,25 @@ use hdk::globals::G_MEM_STACK;
 use serde_json;
 
 zome_functions! {
+    get_an_address: |post_hash: String| {
+        match hdk::get_entry(post_hash) {
+            Err(hdk_error) => hdk_error.to_json(),
+            Ok(maybe_entry) => match maybe_entry {
+                None => json!({}),
+                Some(entry) => match serde_json::from_str(&entry) {
+                    Err(err) => json!({"error deserializing post": err.to_string()}),
+                    Ok(post) => {
+                        // Post is an "entry", so give it to get_entry_address
+                        let maybe_address = hdk::hash_entry(post);
+                        match maybe_address {
```

Comment:
I would reorder some of these with an ok, then Err. Just makes it easier to read imo.

Also part of me thinks we can churn down some of the code using an and_then clause but definitely not needed.

Reply:
Connor made the code according to [this discussion](https://github.com/holochain/app-spec-rust/pull/20).
Nico said he didn't find a better way with combinators.

Reply:
yeah hehehe. Personally I am loosely interested in the discussion of rust syntax stuffs, but verry interested in the specific API for `hash_entry`